packages/core/redux/assetsManager/actions.js

```diff
@@ -428,21 +423,39 @@ export const watchInitTokens = () => async (dispatch, getState) => {
     const tokens = getState().get(DUCK_TOKENS)
     const eventSymbol = web3Converter.bytesToString(data.returnValues.symbol)
 
+    const updateToken = tokens.list().find((t) => t.symbol() === eventSymbol)
```

Comment:
I suggest to replace whole method assetCallback with the following code to improve readability:
```
  const assetCallback = async (data) => {
    const tokens = getState().get(DUCK_TOKENS)
    const eventSymbol = web3Converter.bytesToString(data.returnValues.symbol)

    const token = tokens
      .list()
      .find((t) => t.symbol() === eventSymbol)
      .isFetching(false)
      .isFetched(true)
      .isPending(false)
      .setAddress(data.returnValues.self.toLowerCase())
      .transactionHash(null)

    await dispatch({
      type: TOKENS_UPDATE,
      token
    })

    const assetsList = getState().get(DUCK_ASSETS_MANAGER).assets()
    const assets = Object.entries(assetsList)
      .filter(([key, asset]) => {
        return asset.symbol !== eventSymbol
      })
      .reduce((accumulator, [key, asset]) => {
        accumulator[key] = asset
      }, {})

    const address = data.returnValues.self.toLowerCase()
    assets[address] = {
      self: address,
      by: data.returnValues.by.toLowerCase(),
      event: data.event,
      includedIn: {
        blockNumber: data.blockNumber,
        logIndex: data.logIndex,
        txIndex: data.txIndex,
      },
      address: address,
      platform: data.returnValues.platform.toLowerCase(),
      symbol: data.returnValues.symbol,
      token: data.returnValues.token.toLowerCase(),
      totalSupply: +updateToken.totalSupply(),
    }

    dispatch({ type: SET_ASSETS, payload: { assets } })
  }
```